### ISSUE TITLE
Compiler crashes for worker declaration without a name

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#28419]
**Description:**
$title.

**Steps to reproduce:**
```ballerina
public function main() {
    worker {
    }
}
```

```log
[2021-02-03 12:31:48,790] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.handleDeclaredWithVar(SemanticAnalyzer.java:1066)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:719)
	at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:53)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3585)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3553)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3545)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1872)
	at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:46)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3585)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3553)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:3549)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:436)
	at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:58)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3585)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:408)
	at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:73)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3585)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3553)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3545)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:303)
	at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:163)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:268)
	at io.ballerina.projects.internal.CompilerPhaseRunner.typeCheck(CompilerPhaseRunner.java:217)
	at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:115)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:373)
	at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
	at io.ballerina.projects.ModuleCompilationState$1.generatePlatformSpecificCode(ModuleCompilationState.java:53)
	at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:326)
	at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:136)
	at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:122)
	at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:98)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
	at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:135)
	at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:97)
	at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:69)
	at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
	at io.ballerina.cli.cmd.RunCommand.execute(RunCommand.java:155)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.ballerina.cli.launcher.Main.main(Main.java:58)
```

**Affected Versions:**
slalpha1

**Comment by prabushi** [comment]
The same error comes with the following code. 
Have the following service, and try to add **var** at ** location.

```
import ballerina/http;
import rest_service.records;
import ballerina/log;
import rest_service.mysql;

service /movies on new http:Listener(9090) {
    resource function get view(http:Caller caller, http:Request req) {
        records:Movie[] movies = mysql:query();
        http:Response res = new;
        var // ** Try to add this var after adding rest of the code.
        json|error respJson = movies.cloneWithType(json);
        if (respJson is json) {
            res.setJsonPayload(respJson);
        }
        var result = caller->respond(res);
        if (result is error) {
            log:printError("Error when responding", err = result);
        }
    }
}

```

**Comment by lochana-chathura** [comment]
Fixed in #28973
